src/Service/VendorService/CsvReaderService.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Service\VendorService;
+
+use Iterator;
+
+/**
+ * Class CsvReaderService.
+ */
+class CsvReaderService
+{
+    /**
+     * Read CSV file one line a the time.
+     *
+     * @param $filename
+     *   The file to read
+     *
+     * @return Iterator
+     *   Will yield one line at a time
+     */
+    public function read(string $filename, string $separator = ','): Iterator
```

Comment:
This could be a static function. No reason to pass `CsvReaderService` as a service